### DISCUSSION TITLE
Testing modals?

### DISCUSSION THREAD

**Discussion by sinnbeck** [filamentphp/filament#8043] [Help] [answered]
Is there some clever way of testing modals? I have a modal that pops up when a button in a table is clicked, which has a form. I want to test this form to ensure it works as expected.

This is my code so far (which isnt much)
```
livewire(ApplicationList::class)
        ->callTableAction('reject-da', $application);

**Comment by danharrin** [comment]
What about the form do you want to test?

**Reply by sinnbeck** [reply]
The form shows a editable preview of an email so I want to ensure that's It's sent. And if I edit it, I want to ensure the changes are sent instead of the original text

**Reply by danharrin** [reply, marked as answer]
You can pass form data to `callTableAction()`

**Reply by sinnbeck** [reply]
Yeah but does that submit the modal? I have a test to check if the mail is sent but it isn't passing. I was considering if validation failed but wasn't sure if can test that

**Reply by sinnbeck** [reply]
Ah nevermind. It did work. I had a typo in my assertion :+1: 

In case anyone struggles with it, this is the full test
```php

it('can send refusal email', function () {
    $this->seed(EmailTemplateSeeder::class);
    Mail::fake();

    $application = Application::factory()
        ->has(ApplicationFile::factory()->count(2), 'files')
        ->create([
            'status' => 'pending',
        ]);
    livewire(ApplicationList::class)
        ->callTableAction('reject-da', $application, [
            'subject' => 'Some subject',
            'body' => 'And some text',
        ]);

    Mail::assertSent(ResponseEmail::class);

    expect($application->refresh()->status)
        ->toBe('reject')
        ->and(Email::first())
        ->subject->toBe('Some subject')
        ->body->toBe('And some text');

});
